addons/auth_totp/tests/test_totp.py

```diff
@@ -81,3 +83,61 @@ def test_totp_administration(self):
         self.start_tour('/web', 'totp_tour_setup', login='demo')
         self.start_tour('/web', 'totp_admin_disables', login='admin')
         self.start_tour('/', 'totp_login_disabled', login=None)
+
+    def _build_payload(self, params={}):
+        """
+        Helper to properly build jsonrpc payload
+        """
+        return {
+            'jsonrpc': '2.0',
+            'method': 'call',
+            'id': str(uuid4()),
+            'params': params,
+        }
+
+    def test_totp_mobile_login(self):
+        headers = {
+            'Content-Type': 'application/json',
+        }
+        # 1. Enable 2FA
+        self.start_tour('/web', 'totp_tour_setup', login='demo')
+        self.url_open('/web/session/logout')
+
+        # 2. query login
+        payload = self._build_payload({
+            'db': get_db_name(),
+            'login': 'demo',
+            'password': 'demo',
+            'context': {},
+        })
+        response = self.url_open('/web/session/authenticate', data=json.dumps(payload), headers=headers)
```

Comment:
`json` kwarg does the conversion on its own and automatically sets the request's content-type.